src/test/java/teammates/test/cases/common/UrlTest.java

```diff
@@ -16,4 +18,122 @@ public void testTrimTrailingSlash(){
         assertEquals("abc.com", Url.trimTrailingSlash("abc.com"));
     }
 
+    @Test
+    public void testToString() {
+        
+        ______TS("local file URL (no clear distinction on its base URL)");
+        
+        Url url = new Url("file:///C:/path/to/file.ext");
+        assertEquals("/C:/path/to/file.ext", url.toString());
+        assertEquals("file:///C:/path/to/file.ext", url.toAbsoluteString());
+        
+        ______TS("web URL with no relative path");
+        
+        url = new Url("http://www.google.com");
+        assertEquals("", url.toString());
+        assertEquals("http://www.google.com", url.toAbsoluteString());
+        
+        ______TS("typical web URL");
+        
+        url = new Url("http://www.google.com/page?key1=value1");
+        assertEquals("/page?key1=value1", url.toString());
+        assertEquals("http://www.google.com/page?key1=value1", url.toAbsoluteString());
+        
+        ______TS("malformed URL: no protocol");
+        
+        try {
+            url = new Url("www.google.com/page");
+            signalFailureToDetectException();
+        } catch (AssertionError ae) {
+            ignoreExpectedException();
+        }
+        
+        ______TS("malformed URL: unknown protocol");
+        
+        try {
+            url = new Url("randomprotocol://www.google.com/page");
+            signalFailureToDetectException();
+        } catch (AssertionError ae) {
+            ignoreExpectedException();
+        }
```

Comment:
It's ok to have both these test cases, but they test the same thing. i.e. they go through the same execution path in our code.